+sw_tests/+unit_tests/unittest_spinw.m

```diff
@@ -0,0 +1,119 @@
+classdef unittest_spinw < matlab.mock.TestCase
+    % Runs through unit tests for @spinw/spinw.m
+    properties (TestParameter)
+        spinw_struct_input = { ...
+            {struct('lattice', struct('angle', [pi, pi, (2*pi)/3], ...
+                                     'lat_const', [2, 2, 4])),
+             'spinw_from_struct_lat224.mat'}, ...
+            {struct('lattice', struct('angle', [pi; pi; (2*pi)/3], ...
+                                     'lat_const', [2; 2; 4])), ...
+             'spinw_from_struct_lat224.mat'}};
+        spinw_figure_input = { ...
+            {'default_structure.fig', 'spinw_default.mat'}, ...
+            {'afm_chain_spec.fig', 'spinw_afm_chain.mat'}}
+        spinw_file_input = { ...
+            {'YFeO3_mcphase.cif', 'spinw_from_cif_YFeO3_mcphase.mat'}, ...
+            {'LaFeO3_fullprof.cif', 'spinw_from_cif_LaFeO3_fullprof.mat'}, ...
+            {'BiMn2O5.fst', 'spinw_from_fst_BiMn2O5.mat'}};
+    end
+    methods (Static)
+        function udir = get_unit_test_dir()
+            udir = fullfile('.', 'test_data', 'unit_tests');
+        end
+    end
+    methods
+        function obj = load_spinw(testCase, filename)
+            obj = load(fullfile(testCase.get_unit_test_dir(), 'spinw', filename));
+            obj = obj.data;
+        end
+        function obj = load_figure(testCase, filename)
+            path = fullfile(testCase.get_unit_test_dir(), 'Figure', filename);
+            obj = openfig(path, 'invisible');
+        end
+        function verify_obj(testCase, expected_obj, actual_obj, rel_tol, abs_tol)
+            if nargin < 5
```

Comment:
Atm `verify_val` is only called from `verify_obj`, so we can probably just have the default argument values for tolerance s in `verify_obj` and avoid some duplication (though in another comment I suggest putting this as a separate utility function)

Reply:
It looks like MATLAB testing library has a struct comparator class, would that do an equivalent job to `verify_obj1 in this case?
https://uk.mathworks.com/help/matlab/ref/matlab.unittest.constraints.structcomparator-class.html